src/darwin/Framework/CHIP/MTRSetupPayload.mm

```diff
@@ -252,13 +270,13 @@ - (void)encodeWithCoder:(NSCoder *)coder
     [coder encodeObject:self.version forKey:MTRSetupPayloadCodingKeyVersion];
     [coder encodeObject:self.vendorID forKey:MTRSetupPayloadCodingKeyVendorID];
     [coder encodeObject:self.productID forKey:MTRSetupPayloadCodingKeyProductID];
-    // Casts are safe because commissioning flow and hasShortDiscriminator
-    // values are both pretty small and non-negative.
+    // Casts are safe because commissioning flow, discoveryCapabilities, and
+    // hasShortDiscriminator values are all pretty small and non-negative.
     [coder encodeInteger:static_cast<NSInteger>(self.commissioningFlow) forKey:MTRSetupPayloadCodingKeyCommissioningFlow];
-    [coder encodeObject:self.rendezvousInformation forKey:MTRSetupPayloadCodingKeyRendezvousFlags];
```

Comment:
We'll need to be compatible with the old format, and possible encode so that old software can read this new version. Given that these changes seem largely cosmetic, the simplest way would probably be to keep the values of the string constants the same, and keep encoding `MTRSetupPayloadCodingKeyDiscoveryCapabilities` as an NSNumber object.

Reply:
Done.